[ENH] seedmap functionality

tbweng opened the pull request:
Fixes #98 and fixes #104 .

Changes proposed in this pull request:
- Replace `firstlevelseeding_parallel.m` with `3dTcorr1D`
- Replace several for-loops with functions (e.g., warp seed to native, correlation mapping, warping corrmap to MNI).

tbweng commented:
note: forced push because removing freesurfer is outside scope of this PR. Will create a separate PR for addressing removal of freesurfer.

jdkent commented on seedVoxelCorrelation.sh:
```diff
@@ -243,7 +113,7 @@ for roi in "${roiList[@]}"; do
 		echo "......Mapping $roiName from MNI (standard) to subject EPI (func) space"
 		# Source MNI to EPI warp file
-		MNItoEPIWarp=$(find ${rawEpiDir}/EPItoT1optimized* -maxdepth 1 -name "MNItoEPI_warp.nii.gz")
+		export MNItoEPIWarp=$(find ${rawEpiDir}/EPItoT1optimized* -maxdepth 1 -name "MNItoEPI_warp.nii.gz")
```
I see that exports are pervasive in the code, but what is the purpose of making this variable accessible in the environment (e.g. outside of this script)?

By default variables defined in functions are "global" (e.g. accessible everywhere else in the script) [at least in bash](https://likegeeks.com/bash-functions/#Processing-Variables-in-Bash-Functions).

tbweng replied:
Thanks for the feedback, James! The primary reason for exporting variables was to [make them visible to the child processes spawned by gnu parallel](https://superuser.com/questions/1222086/gnu-parallel-global-variables-and-function).

jdkent replied:
alright, that checks out for me, I approve.

mwvoss commented:
I added a minor edit for an error I was getting in checks for the compcor flag. Otherwise, when I run the native maps come out but the standard are not. I don't have time to work on code today, so I'll just pass on the command and terminal output for reference:

What I ran:
```
docker run -it \
-v /Volumes/vosslabhpc/Projects/CREST:/data \
-v /Users/mwvoss/bin/RestingState:/opt/RestingState \
--entrypoint /opt/RestingState/seedVoxelCorrelation.sh \
hbclab/restingstate:v0.3.1 \
    --epi=/data/derivatives/rsOut/sub-AMBI005/nuisanceRegression/compcor_global/nonfiltered_smooth_data_bp_res4d_normandscaled.nii.gz \
    --motionscrub \
    --roiList=/data/derivatives/rois/roi_list.txt \
    --compcor_global \
    --seedmaps
```

Terminal out (note the _native corrmap files get made but standard do not
```Running /opt/RestingState/seedVoxelCorrelation.sh ...
...Transforming ROIs to EPI space
Academic tradition requires you to cite works you base your article on.
When using programs that use GNU Parallel to process data for publication
please cite:

  O. Tange (2011): GNU Parallel - The Command-Line Power Tool,
  ;login: The USENIX Magazine, February 2011:42-47.

This helps funding further development; AND IT WON'T COST YOU A CENT.
If you pay 10000 EUR you should feel free to use GNU Parallel without citing.

To silence this citation notice: run 'parallel --citation'.


Computers / CPU cores / Max jobs to run
1:local / 4 / 3

Computer:jobs running/jobs completed/%of started jobs/Average seconds to complete
local:3/0/100%/0.0s How did you get here?
AH_MTL_HF_Rhemi has already been mapped from MNI to EPI
roimask: /data/derivatives/rsOut/sub-AMBI005/seedCorrelation/compcor_global/rois/AH_MTL_HF_Rhemi_mask.nii.gz
How did you get here?
How did you get here?
local:2/1/100%/8.0s How did you get here?
AH_MTL_HF has already been mapped from MNI to EPI
roimask: /data/derivatives/rsOut/sub-AMBI005/seedCorrelation/compcor_global/rois/AH_MTL_HF_mask.nii.gz
How did you get here?
How did you get here?
local:1/2/100%/4.0s How did you get here?
UIUC_DMN_core has already been mapped from MNI to EPI
roimask: /data/derivatives/rsOut/sub-AMBI005/seedCorrelation/compcor_global/rois/UIUC_DMN_core_mask.nii.gz
How did you get here?
How did you get here?
local:0/3/100%/2.7s 
...QC Image Setup
Academic tradition requires you to cite works you base your article on.
When using programs that use GNU Parallel to process data for publication
please cite:

  O. Tange (2011): GNU Parallel - The Command-Line Power Tool,
  ;login: The USENIX Magazine, February 2011:42-47.

This helps funding further development; AND IT WON'T COST YOU A CENT.
If you pay 10000 EUR you should feel free to use GNU Parallel without citing.

To silence this citation notice: run 'parallel --citation'.


Computers / CPU cores / Max jobs to run
1:local / 4 / 3

Computer:jobs running/jobs completed/%of started jobs/Average seconds to complete
local:3/0/100%/0.0s UIUC_DMN_core
UIUC_DMN_core QC already exists
local:2/1/100%/1.0s AH_MTL_HF_Rhemi
AH_MTL_HF_Rhemi QC already exists
local:1/2/100%/0.5s AH_MTL_HF
AH_MTL_HF QC already exists
local:0/3/100%/0.3s 
AH_MTL_HF_Rhemi QC already exists
AH_MTL_HF QC already exists
UIUC_DMN_core QC already exists
no seeds to QC
...Seed Voxel Correlation Setup
/opt/RestingState/seedVoxelCorrelation.sh: line 639: [: /data/derivatives/rsOut/sub-AMBI005/seedCorrelation/compcor_global/seedmaps/AH_MTL_HF_Rhemi_ms: binary operator expected
/opt/RestingState/seedVoxelCorrelation.sh: line 639: [: /data/derivatives/rsOut/sub-AMBI005/seedCorrelation/compcor_global/seedmaps/AH_MTL_HF_ms: binary operator expected
/opt/RestingState/seedVoxelCorrelation.sh: line 639: [: /data/derivatives/rsOut/sub-AMBI005/seedCorrelation/compcor_global/seedmaps/UIUC_DMN_core_ms: binary operator expected
...Correlating Seeds With Time Series Data
Academic tradition requires you to cite works you base your article on.
When using programs that use GNU Parallel to process data for publication
please cite:

  O. Tange (2011): GNU Parallel - The Command-Line Power Tool,
  ;login: The USENIX Magazine, February 2011:42-47.

This helps funding further development; AND IT WON'T COST YOU A CENT.
If you pay 10000 EUR you should feel free to use GNU Parallel without citing.

To silence this citation notice: run 'parallel --citation'.

++ 3dTcorr1D: AFNI version=Debian-18.0.05+git24-gb25b21054~dfsg.1-1~nd90+1 (Jan 23 2018) [64-bit]
 + reading dataset file /data/derivatives/rsOut/sub-AMBI005/nuisanceRegression/compcor_global/nonfiltered_smooth_data_bp_res4d_normandscaled.nii.gz
 + reading 1D file /data/derivatives/rsOut/sub-AMBI005/seedCorrelation/compcor_global/rois/UIUC_DMN_core_residvol_ts.txt
 + loading dataset /data/derivatives/rsOut/sub-AMBI005/nuisanceRegression/compcor_global/nonfiltered_smooth_data_bp_res4d_normandscaled.nii.gz into memory
++ Start correlations: 151552 voxels X 1 time series(298); 1 threads
*+ WARNING: THD_Tcorr1D: 124633 voxels skipped because were constant in time
++ Wrote dataset: /data/derivatives/rsOut/sub-AMBI005/seedCorrelation/compcor_global/seedmaps/UIUC_DMN_core_corrmap_native+orig.BRIK
++ 3dTcorr1D: AFNI version=Debian-18.0.05+git24-gb25b21054~dfsg.1-1~nd90+1 (Jan 23 2018) [64-bit]
 + reading dataset file /data/derivatives/rsOut/sub-AMBI005/nuisanceRegression/compcor_global/nonfiltered_smooth_data_bp_res4d_normandscaled.nii.gz
 + reading 1D file /data/derivatives/rsOut/sub-AMBI005/seedCorrelation/compcor_global/rois/AH_MTL_HF_Rhemi_residvol_ts.txt
 + loading dataset /data/derivatives/rsOut/sub-AMBI005/nuisanceRegression/compcor_global/nonfiltered_smooth_data_bp_res4d_normandscaled.nii.gz into memory
++ Start correlations: 151552 voxels X 1 time series(298); 1 threads
*+ WARNING: THD_Tcorr1D: 124633 voxels skipped because were constant in time
++ Wrote dataset: /data/derivatives/rsOut/sub-AMBI005/seedCorrelation/compcor_global/seedmaps/AH_MTL_HF_Rhemi_corrmap_native+orig.BRIK
++ 3dTcorr1D: AFNI version=Debian-18.0.05+git24-gb25b21054~dfsg.1-1~nd90+1 (Jan 23 2018) [64-bit]
 + reading dataset file /data/derivatives/rsOut/sub-AMBI005/nuisanceRegression/compcor_global/nonfiltered_smooth_data_bp_res4d_normandscaled.nii.gz
 + reading 1D file /data/derivatives/rsOut/sub-AMBI005/seedCorrelation/compcor_global/rois/AH_MTL_HF_residvol_ts.txt
 + loading dataset /data/derivatives/rsOut/sub-AMBI005/nuisanceRegression/compcor_global/nonfiltered_smooth_data_bp_res4d_normandscaled.nii.gz into memory
++ Start correlations: 151552 voxels X 1 time series(298); 1 threads
*+ WARNING: THD_Tcorr1D: 124633 voxels skipped because were constant in time
++ Wrote dataset: /data/derivatives/rsOut/sub-AMBI005/seedCorrelation/compcor_global/seedmaps/AH_MTL_HF_corrmap_native+orig.BRIK
Academic tradition requires you to cite works you base your article on.
When using programs that use GNU Parallel to process data for publication
please cite:

  O. Tange (2011): GNU Parallel - The Command-Line Power Tool,
  ;login: The USENIX Magazine, February 2011:42-47.

This helps funding further development; AND IT WON'T COST YOU A CENT.
If you pay 10000 EUR you should feel free to use GNU Parallel without citing.

To silence this citation notice: run 'parallel --citation'.

++ 3dAFNItoNIFTI: AFNI version=Debian-18.0.05+git24-gb25b21054~dfsg.1-1~nd90+1 (Jan 23 2018) [64-bit]
++ 3dAFNItoNIFTI: AFNI version=Debian-18.0.05+git24-gb25b21054~dfsg.1-1~nd90+1 (Jan 23 2018) [64-bit]
++ 3dAFNItoNIFTI: AFNI version=Debian-18.0.05+git24-gb25b21054~dfsg.1-1~nd90+1 (Jan 23 2018) [64-bit]
Academic tradition requires you to cite works you base your article on.
When using programs that use GNU Parallel to process data for publication
please cite:

  O. Tange (2011): GNU Parallel - The Command-Line Power Tool,
  ;login: The USENIX Magazine, February 2011:42-47.

This helps funding further development; AND IT WON'T COST YOU A CENT.
If you pay 10000 EUR you should feel free to use GNU Parallel without citing.

To silence this citation notice: run 'parallel --citation'.

Academic tradition requires you to cite works you base your article on.
When using programs that use GNU Parallel to process data for publication
please cite:

  O. Tange (2011): GNU Parallel - The Command-Line Power Tool,
  ;login: The USENIX Magazine, February 2011:42-47.

This helps funding further development; AND IT WON'T COST YOU A CENT.
If you pay 10000 EUR you should feel free to use GNU Parallel without citing.

To silence this citation notice: run 'parallel --citation'.

...Creating zstat Results Directory
Academic tradition requires you to cite works you base your article on.
When using programs that use GNU Parallel to process data for publication
please cite:

  O. Tange (2011): GNU Parallel - The Command-Line Power Tool,
  ;login: The USENIX Magazine, February 2011:42-47.

This helps funding further development; AND IT WON'T COST YOU A CENT.
If you pay 10000 EUR you should feel free to use GNU Parallel without citing.

To silence this citation notice: run 'parallel --citation'.

...Mapping Correlation For UIUC_DMN_core to MNI
Image Exception : #22 :: ERROR: Could not open image /data/derivatives/rsOut/sub-AMBI005/seedCorrelation/compcor_global/seedmaps/UIUC_DMN_core_corrmap_ms_native
terminate called after throwing an instance of 'RBD_COMMON::BaseException'
environment: line 1:   329 Aborted                 applywarp --in=${seedcorrDir}/${roi}_corrmap_ms_native.nii --ref=${preproc}/reg/standard.nii.gz --out=${seedcorrDir}/${roi}_ms_corrmap_standard.nii.gz --warp=${preproc}/reg/example_func2standard_warp.nii.gz --datatype=float
Cannot open volume /data/derivatives/rsOut/sub-AMBI005/seedCorrelation/compcor_global/seedmaps/UIUC_DMN_core_ms_corrmap_standard for reading!
...Mapping Correlation For AH_MTL_HF_Rhemi to MNI
Image Exception : #22 :: ERROR: Could not open image /data/derivatives/rsOut/sub-AMBI005/seedCorrelation/compcor_global/seedmaps/AH_MTL_HF_Rhemi_corrmap_ms_native
terminate called after throwing an instance of 'RBD_COMMON::BaseException'
environment: line 1:   327 Aborted                 applywarp --in=${seedcorrDir}/${roi}_corrmap_ms_native.nii --ref=${preproc}/reg/standard.nii.gz --out=${seedcorrDir}/${roi}_ms_corrmap_standard.nii.gz --warp=${preproc}/reg/example_func2standard_warp.nii.gz --datatype=float
Cannot open volume /data/derivatives/rsOut/sub-AMBI005/seedCorrelation/compcor_global/seedmaps/AH_MTL_HF_Rhemi_ms_corrmap_standard for reading!
...Mapping Correlation For AH_MTL_HF to MNI
Image Exception : #22 :: ERROR: Could not open image /data/derivatives/rsOut/sub-AMBI005/seedCorrelation/compcor_global/seedmaps/AH_MTL_HF_corrmap_ms_native
terminate called after throwing an instance of 'RBD_COMMON::BaseException'
environment: line 1:   328 Aborted                 applywarp --in=${seedcorrDir}/${roi}_corrmap_ms_native.nii --ref=${preproc}/reg/standard.nii.gz --out=${seedcorrDir}/${roi}_ms_corrmap_standard.nii.gz --warp=${preproc}/reg/example_func2standard_warp.nii.gz --datatype=float
Cannot open volume /data/derivatives/rsOut/sub-AMBI005/seedCorrelation/compcor_global/seedmaps/AH_MTL_HF_ms_corrmap_standard for reading!
Academic tradition requires you to cite works you base your article on.
When using programs that use GNU Parallel to process data for publication
please cite:

  O. Tange (2011): GNU Parallel - The Command-Line Power Tool,
  ;login: The USENIX Magazine, February 2011:42-47.

This helps funding further development; AND IT WON'T COST YOU A CENT.
If you pay 10000 EUR you should feel free to use GNU Parallel without citing.

To silence this citation notice: run 'parallel --citation'.

chmod: cannot access '': No such file or directory
/opt/RestingState/seedVoxelCorrelation.sh Complete
Please make sure that the ROI folders were created in the /data/derivatives/rsOut/sub-AMBI005/seedCorrelation/compcor_global/rois/ folder.
If resultant warped seeds (to MNI) were too small, they were NOT processed.  Check /data/derivatives/rsOut/sub-AMBI005/seedsTooSmall for exclusions.
If motionscrubbing was set to 1 or 2, make sure that motionscrubbed data was created.
OCTAVE/Matlab wouldn't give an error even if this step was not successfully done.
```

jdkent commented:
Ah, sorry I forgot there was a new command line option being introduced. The circleci should be modified. I suggest adding it to the `compcor` [test here](https://github.com/HBClab/RestingState/blob/1ddc7a682f3e574e0b5427b7d204362fa50575c1/.circleci/config.yml#L120). The text file may also need to be changed to accommodate the new outputs.

mwvoss commented:
Yeah, it will be common usage to run seedVoxelCorrelation within the wrapper without seedmaps option (many ROIs for cross-correlation output), and then on its own with the --seedmaps option for voxelwise analyses on a subset of ROIs

tbweng commented:
> I added a minor edit for an error I was getting in checks for the compcor flag. Otherwise, when I run the native maps come out but the standard are not. I don't have time to work on code today, so I'll just pass on the command and terminal output for reference:
> 

@mwvoss I fixed a few typos that was causing your errors with warping the corrmap to MNI. It should work with the latest commit!

mwvoss commented:
Closer...though something still seems off with the loop at the end. I'm getting the standard corrmap for one seed but not others in the list. Output on the server in Projects/CREST/derivatives/rsOut/sub-AMBI006/seedCorrelation/..

See towards the end of the terminal output..
```
bash-3.2$ docker run -it \
> -v /Volumes/vosslabhpc/Projects/CREST:/data \
> -v /Users/mwvoss/bin/RestingState:/opt/RestingState \
> --entrypoint /opt/RestingState/seedVoxelCorrelation.sh \
> hbclab/restingstate:v0.3.1 \
>     --epi=/data/derivatives/rsOut/sub-AMBI006/nuisanceRegression/compcor_global/nonfiltered_smooth_data_bp_res4d_normandscaled.nii.gz \
>     --motionscrub \
>     --roiList=/data/derivatives/rois/roi_list.txt \
>     --compcor_global \
>     --seedmaps
Running /opt/RestingState/seedVoxelCorrelation.sh ...
...Transforming ROIs to EPI space
Academic tradition requires you to cite works you base your article on.
When using programs that use GNU Parallel to process data for publication
please cite:

  O. Tange (2011): GNU Parallel - The Command-Line Power Tool,
  ;login: The USENIX Magazine, February 2011:42-47.

This helps funding further development; AND IT WON'T COST YOU A CENT.
If you pay 10000 EUR you should feel free to use GNU Parallel without citing.

To silence this citation notice: run 'parallel --citation'.


Computers / CPU cores / Max jobs to run
1:local / 4 / 3

Computer:jobs running/jobs completed/%of started jobs/Average seconds to complete
local:3/0/100%/0.0s How did you get here?
UIUC_DMN_core has already been mapped from MNI to EPI
roimask: /data/derivatives/rsOut/sub-AMBI006/seedCorrelation/compcor_global/rois/UIUC_DMN_core_mask.nii.gz
How did you get here?
How did you get here?
local:2/1/100%/10.0s How did you get here?
AH_MTL_HF_Rhemi has already been mapped from MNI to EPI
roimask: /data/derivatives/rsOut/sub-AMBI006/seedCorrelation/compcor_global/rois/AH_MTL_HF_Rhemi_mask.nii.gz
How did you get here?
How did you get here?
local:1/2/100%/5.0s How did you get here?
AH_MTL_HF has already been mapped from MNI to EPI
roimask: /data/derivatives/rsOut/sub-AMBI006/seedCorrelation/compcor_global/rois/AH_MTL_HF_mask.nii.gz
How did you get here?
How did you get here?
local:0/3/100%/3.3s 
...QC Image Setup
Academic tradition requires you to cite works you base your article on.
When using programs that use GNU Parallel to process data for publication
please cite:

  O. Tange (2011): GNU Parallel - The Command-Line Power Tool,
  ;login: The USENIX Magazine, February 2011:42-47.

This helps funding further development; AND IT WON'T COST YOU A CENT.
If you pay 10000 EUR you should feel free to use GNU Parallel without citing.

To silence this citation notice: run 'parallel --citation'.


Computers / CPU cores / Max jobs to run
1:local / 4 / 3

Computer:jobs running/jobs completed/%of started jobs/Average seconds to complete
local:3/0/100%/0.0s AH_MTL_HF
AH_MTL_HF QC already exists
local:2/1/100%/0.0s AH_MTL_HF_Rhemi
AH_MTL_HF_Rhemi QC already exists
local:1/2/100%/0.0s UIUC_DMN_core
UIUC_DMN_core QC already exists
local:0/3/100%/0.0s 
UIUC_DMN_core QC already exists
AH_MTL_HF_Rhemi QC already exists
AH_MTL_HF QC already exists
no seeds to QC
...Seed Voxel Correlation Setup
...Correlating Seeds With Time Series Data
Academic tradition requires you to cite works you base your article on.
When using programs that use GNU Parallel to process data for publication
please cite:

  O. Tange (2011): GNU Parallel - The Command-Line Power Tool,
  ;login: The USENIX Magazine, February 2011:42-47.

This helps funding further development; AND IT WON'T COST YOU A CENT.
If you pay 10000 EUR you should feel free to use GNU Parallel without citing.

To silence this citation notice: run 'parallel --citation'.

++ 3dTcorr1D: AFNI version=Debian-18.0.05+git24-gb25b21054~dfsg.1-1~nd90+1 (Jan 23 2018) [64-bit]
 + reading dataset file /data/derivatives/rsOut/sub-AMBI006/nuisanceRegression/compcor_global/nonfiltered_smooth_data_bp_res4d_normandscaled.nii.gz
*+ WARNING:   If you are performing spatial transformations on an oblique dset, 
  such as /data/derivatives/rsOut/sub-AMBI006/nuisanceRegression/compcor_global/nonfiltered_smooth_data_bp_res4d_normandscaled.nii.gz,
  or viewing/combining it with volumes of differing obliquity,
  you should consider running: 
     3dWarp -deoblique 
  on this and  other oblique datasets in the same session.
 See 3dWarp -help for details.
++ Oblique dataset:/data/derivatives/rsOut/sub-AMBI006/nuisanceRegression/compcor_global/nonfiltered_smooth_data_bp_res4d_normandscaled.nii.gz is 10.769126 degrees from plumb.
 + reading 1D file /data/derivatives/rsOut/sub-AMBI006/seedCorrelation/compcor_global/rois/UIUC_DMN_core_residvol_ts.txt
 + loading dataset /data/derivatives/rsOut/sub-AMBI006/nuisanceRegression/compcor_global/nonfiltered_smooth_data_bp_res4d_normandscaled.nii.gz into memory
++ Start correlations: 151552 voxels X 1 time series(298); 1 threads
*+ WARNING: THD_Tcorr1D: 120108 voxels skipped because were constant in time
++ Wrote dataset: /data/derivatives/rsOut/sub-AMBI006/seedCorrelation/compcor_global/seedmaps/UIUC_DMN_core_corrmap_native+orig.BRIK
++ 3dTcorr1D: AFNI version=Debian-18.0.05+git24-gb25b21054~dfsg.1-1~nd90+1 (Jan 23 2018) [64-bit]
 + reading dataset file /data/derivatives/rsOut/sub-AMBI006/nuisanceRegression/compcor_global/nonfiltered_smooth_data_bp_res4d_normandscaled.nii.gz
*+ WARNING:   If you are performing spatial transformations on an oblique dset, 
  such as /data/derivatives/rsOut/sub-AMBI006/nuisanceRegression/compcor_global/nonfiltered_smooth_data_bp_res4d_normandscaled.nii.gz,
  or viewing/combining it with volumes of differing obliquity,
  you should consider running: 
     3dWarp -deoblique 
  on this and  other oblique datasets in the same session.
 See 3dWarp -help for details.
++ Oblique dataset:/data/derivatives/rsOut/sub-AMBI006/nuisanceRegression/compcor_global/nonfiltered_smooth_data_bp_res4d_normandscaled.nii.gz is 10.769126 degrees from plumb.
 + reading 1D file /data/derivatives/rsOut/sub-AMBI006/seedCorrelation/compcor_global/rois/AH_MTL_HF_residvol_ts.txt
 + loading dataset /data/derivatives/rsOut/sub-AMBI006/nuisanceRegression/compcor_global/nonfiltered_smooth_data_bp_res4d_normandscaled.nii.gz into memory
++ Start correlations: 151552 voxels X 1 time series(298); 1 threads
*+ WARNING: THD_Tcorr1D: 120108 voxels skipped because were constant in time
++ Wrote dataset: /data/derivatives/rsOut/sub-AMBI006/seedCorrelation/compcor_global/seedmaps/AH_MTL_HF_corrmap_native+orig.BRIK
++ 3dTcorr1D: AFNI version=Debian-18.0.05+git24-gb25b21054~dfsg.1-1~nd90+1 (Jan 23 2018) [64-bit]
 + reading dataset file /data/derivatives/rsOut/sub-AMBI006/nuisanceRegression/compcor_global/nonfiltered_smooth_data_bp_res4d_normandscaled.nii.gz
*+ WARNING:   If you are performing spatial transformations on an oblique dset, 
  such as /data/derivatives/rsOut/sub-AMBI006/nuisanceRegression/compcor_global/nonfiltered_smooth_data_bp_res4d_normandscaled.nii.gz,
  or viewing/combining it with volumes of differing obliquity,
  you should consider running: 
     3dWarp -deoblique 
  on this and  other oblique datasets in the same session.
 See 3dWarp -help for details.
++ Oblique dataset:/data/derivatives/rsOut/sub-AMBI006/nuisanceRegression/compcor_global/nonfiltered_smooth_data_bp_res4d_normandscaled.nii.gz is 10.769126 degrees from plumb.
 + reading 1D file /data/derivatives/rsOut/sub-AMBI006/seedCorrelation/compcor_global/rois/AH_MTL_HF_Rhemi_residvol_ts.txt
 + loading dataset /data/derivatives/rsOut/sub-AMBI006/nuisanceRegression/compcor_global/nonfiltered_smooth_data_bp_res4d_normandscaled.nii.gz into memory
++ Start correlations: 151552 voxels X 1 time series(298); 1 threads
*+ WARNING: THD_Tcorr1D: 120108 voxels skipped because were constant in time
++ Wrote dataset: /data/derivatives/rsOut/sub-AMBI006/seedCorrelation/compcor_global/seedmaps/AH_MTL_HF_Rhemi_corrmap_native+orig.BRIK
Academic tradition requires you to cite works you base your article on.
When using programs that use GNU Parallel to process data for publication
please cite:

  O. Tange (2011): GNU Parallel - The Command-Line Power Tool,
  ;login: The USENIX Magazine, February 2011:42-47.

This helps funding further development; AND IT WON'T COST YOU A CENT.
If you pay 10000 EUR you should feel free to use GNU Parallel without citing.

To silence this citation notice: run 'parallel --citation'.

++ 3dAFNItoNIFTI: AFNI version=Debian-18.0.05+git24-gb25b21054~dfsg.1-1~nd90+1 (Jan 23 2018) [64-bit]
*+ WARNING:   If you are performing spatial transformations on an oblique dset, 
  such as /data/derivatives/rsOut/sub-AMBI006/seedCorrelation/compcor_global/seedmaps/AH_MTL_HF_Rhemi_corrmap_native+orig.BRIK,
  or viewing/combining it with volumes of differing obliquity,
  you should consider running: 
     3dWarp -deoblique 
  on this and  other oblique datasets in the same session.
 See 3dWarp -help for details.
++ Oblique dataset:/data/derivatives/rsOut/sub-AMBI006/seedCorrelation/compcor_global/seedmaps/AH_MTL_HF_Rhemi_corrmap_native+orig.BRIK is 10.769126 degrees from plumb.
++ 3dAFNItoNIFTI: AFNI version=Debian-18.0.05+git24-gb25b21054~dfsg.1-1~nd90+1 (Jan 23 2018) [64-bit]
*+ WARNING:   If you are performing spatial transformations on an oblique dset, 
  such as /data/derivatives/rsOut/sub-AMBI006/seedCorrelation/compcor_global/seedmaps/AH_MTL_HF_corrmap_native+orig.BRIK,
  or viewing/combining it with volumes of differing obliquity,
  you should consider running: 
     3dWarp -deoblique 
  on this and  other oblique datasets in the same session.
 See 3dWarp -help for details.
++ Oblique dataset:/data/derivatives/rsOut/sub-AMBI006/seedCorrelation/compcor_global/seedmaps/AH_MTL_HF_corrmap_native+orig.BRIK is 10.769126 degrees from plumb.
++ 3dAFNItoNIFTI: AFNI version=Debian-18.0.05+git24-gb25b21054~dfsg.1-1~nd90+1 (Jan 23 2018) [64-bit]
*+ WARNING:   If you are performing spatial transformations on an oblique dset, 
  such as /data/derivatives/rsOut/sub-AMBI006/seedCorrelation/compcor_global/seedmaps/UIUC_DMN_core_corrmap_native+orig.BRIK,
  or viewing/combining it with volumes of differing obliquity,
  you should consider running: 
     3dWarp -deoblique 
  on this and  other oblique datasets in the same session.
 See 3dWarp -help for details.
++ Oblique dataset:/data/derivatives/rsOut/sub-AMBI006/seedCorrelation/compcor_global/seedmaps/UIUC_DMN_core_corrmap_native+orig.BRIK is 10.769126 degrees from plumb.
Academic tradition requires you to cite works you base your article on.
When using programs that use GNU Parallel to process data for publication
please cite:

  O. Tange (2011): GNU Parallel - The Command-Line Power Tool,
  ;login: The USENIX Magazine, February 2011:42-47.

This helps funding further development; AND IT WON'T COST YOU A CENT.
If you pay 10000 EUR you should feel free to use GNU Parallel without citing.

To silence this citation notice: run 'parallel --citation'.

++ 3dTcorr1D: AFNI version=Debian-18.0.05+git24-gb25b21054~dfsg.1-1~nd90+1 (Jan 23 2018) [64-bit]
 + reading dataset file /data/derivatives/rsOut/sub-AMBI006/nuisanceRegression/compcor_global/nonfiltered_smooth_data_bp_res4d_normandscaled.nii.gz
*+ WARNING:   If you are performing spatial transformations on an oblique dset, 
  such as /data/derivatives/rsOut/sub-AMBI006/nuisanceRegression/compcor_global/nonfiltered_smooth_data_bp_res4d_normandscaled.nii.gz,
  or viewing/combining it with volumes of differing obliquity,
  you should consider running: 
     3dWarp -deoblique 
  on this and  other oblique datasets in the same session.
 See 3dWarp -help for details.
++ Oblique dataset:/data/derivatives/rsOut/sub-AMBI006/nuisanceRegression/compcor_global/nonfiltered_smooth_data_bp_res4d_normandscaled.nii.gz is 10.769126 degrees from plumb.
 + reading 1D file /data/derivatives/rsOut/sub-AMBI006/seedCorrelation/compcor_global/rois/AH_MTL_HF_residvol_ms_ts.txt
 + loading dataset /data/derivatives/rsOut/sub-AMBI006/nuisanceRegression/compcor_global/nonfiltered_smooth_data_bp_res4d_normandscaled.nii.gz into memory
++ Start correlations: 151552 voxels X 1 time series(298); 1 threads
*+ WARNING: THD_Tcorr1D: 120108 voxels skipped because were constant in time
++ Wrote dataset: /data/derivatives/rsOut/sub-AMBI006/seedCorrelation/compcor_global/seedmaps/AH_MTL_HF_corrmap_ms_native+orig.BRIK
++ 3dTcorr1D: AFNI version=Debian-18.0.05+git24-gb25b21054~dfsg.1-1~nd90+1 (Jan 23 2018) [64-bit]
 + reading dataset file /data/derivatives/rsOut/sub-AMBI006/nuisanceRegression/compcor_global/nonfiltered_smooth_data_bp_res4d_normandscaled.nii.gz
*+ WARNING:   If you are performing spatial transformations on an oblique dset, 
  such as /data/derivatives/rsOut/sub-AMBI006/nuisanceRegression/compcor_global/nonfiltered_smooth_data_bp_res4d_normandscaled.nii.gz,
  or viewing/combining it with volumes of differing obliquity,
  you should consider running: 
     3dWarp -deoblique 
  on this and  other oblique datasets in the same session.
 See 3dWarp -help for details.
++ Oblique dataset:/data/derivatives/rsOut/sub-AMBI006/nuisanceRegression/compcor_global/nonfiltered_smooth_data_bp_res4d_normandscaled.nii.gz is 10.769126 degrees from plumb.
 + reading 1D file /data/derivatives/rsOut/sub-AMBI006/seedCorrelation/compcor_global/rois/AH_MTL_HF_residvol_ms_ts.txt
 + loading dataset /data/derivatives/rsOut/sub-AMBI006/nuisanceRegression/compcor_global/nonfiltered_smooth_data_bp_res4d_normandscaled.nii.gz into memory
++ Start correlations: 151552 voxels X 1 time series(298); 1 threads
*+ WARNING: THD_Tcorr1D: 120108 voxels skipped because were constant in time
++ Wrote dataset: /data/derivatives/rsOut/sub-AMBI006/seedCorrelation/compcor_global/seedmaps/AH_MTL_HF_corrmap_ms_native+orig.BRIK
++ 3dTcorr1D: AFNI version=Debian-18.0.05+git24-gb25b21054~dfsg.1-1~nd90+1 (Jan 23 2018) [64-bit]
 + reading dataset file /data/derivatives/rsOut/sub-AMBI006/nuisanceRegression/compcor_global/nonfiltered_smooth_data_bp_res4d_normandscaled.nii.gz
*+ WARNING:   If you are performing spatial transformations on an oblique dset, 
  such as /data/derivatives/rsOut/sub-AMBI006/nuisanceRegression/compcor_global/nonfiltered_smooth_data_bp_res4d_normandscaled.nii.gz,
  or viewing/combining it with volumes of differing obliquity,
  you should consider running: 
     3dWarp -deoblique 
  on this and  other oblique datasets in the same session.
 See 3dWarp -help for details.
++ Oblique dataset:/data/derivatives/rsOut/sub-AMBI006/nuisanceRegression/compcor_global/nonfiltered_smooth_data_bp_res4d_normandscaled.nii.gz is 10.769126 degrees from plumb.
 + reading 1D file /data/derivatives/rsOut/sub-AMBI006/seedCorrelation/compcor_global/rois/AH_MTL_HF_residvol_ms_ts.txt
 + loading dataset /data/derivatives/rsOut/sub-AMBI006/nuisanceRegression/compcor_global/nonfiltered_smooth_data_bp_res4d_normandscaled.nii.gz into memory
++ Start correlations: 151552 voxels X 1 time series(298); 1 threads
*+ WARNING: THD_Tcorr1D: 120108 voxels skipped because were constant in time
++ Wrote dataset: /data/derivatives/rsOut/sub-AMBI006/seedCorrelation/compcor_global/seedmaps/AH_MTL_HF_corrmap_ms_native+orig.BRIK
Academic tradition requires you to cite works you base your article on.
When using programs that use GNU Parallel to process data for publication
please cite:

  O. Tange (2011): GNU Parallel - The Command-Line Power Tool,
  ;login: The USENIX Magazine, February 2011:42-47.

This helps funding further development; AND IT WON'T COST YOU A CENT.
If you pay 10000 EUR you should feel free to use GNU Parallel without citing.

To silence this citation notice: run 'parallel --citation'.

++ 3dAFNItoNIFTI: AFNI version=Debian-18.0.05+git24-gb25b21054~dfsg.1-1~nd90+1 (Jan 23 2018) [64-bit]
*+ WARNING:   If you are performing spatial transformations on an oblique dset, 
  such as /data/derivatives/rsOut/sub-AMBI006/seedCorrelation/compcor_global/seedmaps/AH_MTL_HF_corrmap_ms_native+orig.BRIK,
  or viewing/combining it with volumes of differing obliquity,
  you should consider running: 
     3dWarp -deoblique 
  on this and  other oblique datasets in the same session.
 See 3dWarp -help for details.
++ Oblique dataset:/data/derivatives/rsOut/sub-AMBI006/seedCorrelation/compcor_global/seedmaps/AH_MTL_HF_corrmap_ms_native+orig.BRIK is 10.769126 degrees from plumb.
++ 3dAFNItoNIFTI: AFNI version=Debian-18.0.05+git24-gb25b21054~dfsg.1-1~nd90+1 (Jan 23 2018) [64-bit]
*+ WARNING:   If you are performing spatial transformations on an oblique dset, 
  such as /data/derivatives/rsOut/sub-AMBI006/seedCorrelation/compcor_global/seedmaps/AH_MTL_HF_corrmap_ms_native+orig.BRIK,
  or viewing/combining it with volumes of differing obliquity,
  you should consider running: 
     3dWarp -deoblique 
  on this and  other oblique datasets in the same session.
 See 3dWarp -help for details.
++ Oblique dataset:/data/derivatives/rsOut/sub-AMBI006/seedCorrelation/compcor_global/seedmaps/AH_MTL_HF_corrmap_ms_native+orig.BRIK is 10.769126 degrees from plumb.
++ 3dAFNItoNIFTI: AFNI version=Debian-18.0.05+git24-gb25b21054~dfsg.1-1~nd90+1 (Jan 23 2018) [64-bit]
*+ WARNING:   If you are performing spatial transformations on an oblique dset, 
  such as /data/derivatives/rsOut/sub-AMBI006/seedCorrelation/compcor_global/seedmaps/AH_MTL_HF_corrmap_ms_native+orig.BRIK,
  or viewing/combining it with volumes of differing obliquity,
  you should consider running: 
     3dWarp -deoblique 
  on this and  other oblique datasets in the same session.
 See 3dWarp -help for details.
++ Oblique dataset:/data/derivatives/rsOut/sub-AMBI006/seedCorrelation/compcor_global/seedmaps/AH_MTL_HF_corrmap_ms_native+orig.BRIK is 10.769126 degrees from plumb.
...Creating zstat Results Directory
Academic tradition requires you to cite works you base your article on.
When using programs that use GNU Parallel to process data for publication
please cite:

  O. Tange (2011): GNU Parallel - The Command-Line Power Tool,
  ;login: The USENIX Magazine, February 2011:42-47.

This helps funding further development; AND IT WON'T COST YOU A CENT.
If you pay 10000 EUR you should feel free to use GNU Parallel without citing.

To silence this citation notice: run 'parallel --citation'.

...Mapping Correlation For UIUC_DMN_core to MNI
Image Exception : #22 :: ERROR: Could not open image /data/derivatives/rsOut/sub-AMBI006/seedCorrelation/compcor_global/seedmaps/UIUC_DMN_core_corrmap_ms_native
terminate called after throwing an instance of 'RBD_COMMON::BaseException'
environment: line 1:   353 Aborted                 applywarp --in=${seedcorrDir}/${roi}_corrmap_ms_native.nii --ref=${preproc}/reg/standard.nii.gz --out=${seedcorrDir}/${roi}_corrmap_ms_standard.nii --warp=${preproc}/reg/example_func2standard_warp.nii.gz --datatype=float
Cannot open volume /data/derivatives/rsOut/sub-AMBI006/seedCorrelation/compcor_global/seedmaps/UIUC_DMN_core_corrmap_ms_standard for reading!
...Mapping Correlation For AH_MTL_HF_Rhemi to MNI
Image Exception : #22 :: ERROR: Could not open image /data/derivatives/rsOut/sub-AMBI006/seedCorrelation/compcor_global/seedmaps/AH_MTL_HF_Rhemi_corrmap_ms_native
terminate called after throwing an instance of 'RBD_COMMON::BaseException'
environment: line 1:   355 Aborted                 applywarp --in=${seedcorrDir}/${roi}_corrmap_ms_native.nii --ref=${preproc}/reg/standard.nii.gz --out=${seedcorrDir}/${roi}_corrmap_ms_standard.nii --warp=${preproc}/reg/example_func2standard_warp.nii.gz --datatype=float
Cannot open volume /data/derivatives/rsOut/sub-AMBI006/seedCorrelation/compcor_global/seedmaps/AH_MTL_HF_Rhemi_corrmap_ms_standard for reading!
...Mapping Correlation For AH_MTL_HF to MNI

/opt/RestingState/seedVoxelCorrelation.sh Complete
Please make sure that the ROI folders were created in the /data/derivatives/rsOut/sub-AMBI006/seedCorrelation/compcor_global/rois/ folder.
If resultant warped seeds (to MNI) were too small, they were NOT processed.  Check /data/derivatives/rsOut/sub-AMBI006/seedsTooSmall for exclusions.
If motionscrubbing was set to 1 or 2, make sure that motionscrubbed data was created.
OCTAVE/Matlab wouldn't give an error even if this step was not successfully done.
```

jdkent commented:
If we modify the `.circleci.yml` file, then hopefully we can catch these errors before @mwvoss has to test them. @tbweng do you need any instruction on how to do this?

tbweng commented:
> If we modify the `.circleci/config.yml` file, then hopefully we can catch these errors before @mwvoss has to test them. @tbweng do you need any instruction on how to do this?

@jdkent I will add an extra test with the motionscrub flag.

tbweng commented:
> Closer...though something still seems off with the loop at the end. I'm getting the standard corrmap for one seed but not others in the list. Output on the server in Projects/CREST/derivatives/rsOut/sub-AMBI006/seedCorrelation/..
> 
> See towards the end of the terminal output..

@mwvoss latest commits should resolve that issue. Fixed syntax for passing the `{roi}` variable to gnu parallel.
@jdkent the latest `.circleci/config.yml` file has an additional test for running seedmapping only (with compcor_global and motionscrub flags)

tbweng commented:
> Thanks for adding the test, I fixed a typo and updated the expected outputs, the other thing I would like context on is why we are getting `How did you get here?` output from the `--seedmap` option, it suggests clobber is running into a state it should not be in.

Resolved with latest commit. The `clob=false` needed to be exported and accessible to extractROIts function when called by gnu parallel.

mwvoss commented:
#Thank you @tbweng @jdkent, great to see this working again!  I'm signed off on the PR.

jdkent commented:
Great work @tbweng! and thanks @mwvoss for testing the "work in progress" code! I can merge this in and release v0.3.2 (since it looks like --seedmap was already a feature in the code, I believe this would constitute a bugfix).

tbweng commented:
> Great work @tbweng! and thanks @mwvoss for testing the "work in progress" code! I can merge this in and release v0.3.2 (since it looks like --seedmap was already a feature in the code, I believe this would constitute a bugfix).

^ or enhancement? Accelerated processing time via parallel and replace old seedmapping code with `3dTcorr1D`.
